dll/shellext/cabview/cabview.h

```diff
@@ -0,0 +1,278 @@
+/*
+ * PROJECT:     ReactOS CabView Shell Extension
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * COPYRIGHT:   Whindmar Saksit (whindsaks@proton.me)
+ */
+
+#pragma once
+#include "precomp.h"
+#include "resource.h"
+
+#define FLATFOLDER TRUE
+
+EXTERN_C const GUID CLSID_CabFolder;
+extern HINSTANCE g_hInst;
+
+enum EXTRACTCALLBACKMSG { ECM_BEGIN, ECM_FILE, ECM_PREPAREPATH, ECM_ERROR };
+struct EXTRACTCALLBACKDATA
+{
+    LPCWSTR Path;
+    const FDINOTIFICATION *pfdin;
+    HRESULT hr;
+};
+typedef HRESULT (CALLBACK*EXTRACTCALLBACK)(EXTRACTCALLBACKMSG msg, const EXTRACTCALLBACKDATA &data, LPVOID cookie);
+HRESULT ExtractCabinet(LPCWSTR cab, LPCWSTR destination, EXTRACTCALLBACK callback, LPVOID cookie);
+
+
+#include <pshpack1.h>
+struct CABITEM
+{
+    WORD cb;
+    WORD Unknown; // Not sure what Windows uses this for, we always store 0
+    UINT Size;
+    WORD Date, Time; // DOS
+    WORD Attrib;
+    WORD NameOffset;
+    WCHAR Path[ANYSIZE_ARRAY];
+
+#if FLATFOLDER
```

Comment:
It can be `0` or `!= 0`. On Windows the cab folder is "flat". It would be nice to support subfolders at some point but it requires a fair amount of extra work because the raw CAB API is flat.